x-pack/functionbeat/magefile.go

```diff
@@ -61,11 +97,16 @@ func Package() {
 	start := time.Now()
 	defer func() { fmt.Println("package ran for", time.Since(start)) }()
 
-	devtools.MustUsePackaging("functionbeat", "x-pack/functionbeat/dev-tools/packaging/packages.yml")
-
 	mg.Deps(Update)
 	mg.Deps(CrossBuild, CrossBuildGoDaemon)
-	mg.SerialDeps(devtools.Package, TestPackages)
+	for _, provider := range selectedProviders {
+		devtools.MustUsePackaging("functionbeat", "x-pack/functionbeat/dev-tools/packaging/packages.yml")
+		for _, args := range devtools.Packages {
+			args.Spec.ExtraVar("Provider", provider)
+		}
+
+		mg.SerialDeps(devtools.Package, TestPackages)
```

Comment:
`SerialDeps `will only run those functions one time. Calling them in a loop will have no effect after the first iteration.